documents/coredocument.go

```diff
@@ -528,3 +537,22 @@ func (cd *CoreDocument) Signatures() (signatures []coredocumentpb.Signature) {
 
 	return signatures
 }
+
+func populateVersions(cd *coredocumentpb.CoreDocument, prevCD *coredocumentpb.CoreDocument) (err error) {
+	if prevCD != nil {
+		cd.PreviousVersion = prevCD.CurrentVersion
+		cd.CurrentVersion = prevCD.NextVersion
+		cd.CurrentPreimage = prevCD.NextPreimage
+	} else {
+		cd.CurrentPreimage, cd.CurrentVersion, err = utils.GenerateHashPair(idSize)
+		cd.DocumentIdentifier = cd.CurrentVersion
```

Comment:
What about cd.PreviousVersion and cd.previousHash. Can we assume the are nil or 0x here?

Reply:
its nil